File: frontend/src/services/TodoService.js
import api from './api';

const TodoService = {
  getTodos: async () => {
    console.log('Fetching todos...');
    const response = await api.get('/todos');
    console.log('Todos received:', response.data);
    return response.data;
  },

  addTodo: async (todoData) => {
    console.log('Adding todo:', todoData);
    const response = await api.post('/todos', todoData);
    console.log('Todo added response:', response.data);
    return response.data;
  },

  updateTodo: async (id, todoData) => {
    console.log('Updating todo:', id, todoData);
    const response = await api.put(`/todos/${id}`, todoData);
    return response.data;
  },

  deleteTodo: async (id) => {
    console.log('Deleting todo:', id);
    await api.delete(`/todos/${id}`);
  },

  toggleTodo: async (id) => {
    console.log('Toggling todo:', id);
    await api.patch(`/todos/${id}/toggle`);
  }
};

export default TodoService;

File: backend/src/main/java/com/obs/backend/service/TodoService.java
package com.obs.backend.service;

import com.obs.backend.dto.student.TodoDTO;

import java.util.List;

public interface TodoService {
    List<TodoDTO> getTodos(String username);
    TodoDTO addTodo(String username, TodoDTO todoDTO);
    TodoDTO updateTodo(Long id, TodoDTO todoDTO);
    void deleteTodo(Long id);
    void toggleTodo(Long id);
}
